util/abs/provision_pe_xl_nodes.rb

```diff
@@ -1,9 +1,83 @@
 # frozen_string_literal: true
 
+# TODO: make this a Bolt task once abs_helper is available as a gem
+
+require "optparse"
 require "yaml"
+require "json"
 require "./setup/helpers/abs_helper.rb"
 include AbsHelper # rubocop:disable Style/MixinUsage
 
+DESCRIPTION = <<~DESCRIPTION
+  This script was created to assist in working with the pe_xl module (https://github.com/reidmv/reidmv-pe_xl).
+  It provisions the nodes used by the module and generates the Bolt inventory and parameter files populated with the provisioned hosts.
+
+  * Note: Because the script is designed to work with GPLT it sets up the environment without HA by default.
+
+  EC2 hosts are provisioned for the following roles:
+  * Core roles:
+   - master
+   - puppet_db
+   - compiler_a
+   - compiler_b
+
+  * HA roles:
+   - master_replica
+   - puppet_db_replica
+
+DESCRIPTION
+
+options = {}
+
+# Note: looks like 'Store options to a Hash' doesn't work in Ruby 2.3.0.
+# https://ruby-doc.org/stdlib-2.6.3/libdoc/optparse/rdoc/OptionParser.html
+#  `end.parse!(into: options)`
+# TODO: update to use '(into: options)' after Ruby update
+# TODO: error when invalid options are specified
+# TODO: re-order the options?
+OptionParser.new do |opts|
+  opts.banner = "Usage: provision_pe_xl_nodes.rb [options]"
+
+  opts.on("-h", "--help", "Display the help text") do
+    puts DESCRIPTION
+    puts opts
+    puts
+    exit
+  end
+
+  opts.on("--ha", "Specifies that the environment should be set up for HA")
+
+  # TODO: noop and test options are mutually exclusive;
+  # error when both are specified?
```

Comment:
are they?  can't you do a noop using the test data?

Reply:
Well, you can specify them together:
```
bill.claytor:~/RubymineProjects/_forks/gatling-puppet-load-test> ruby util/abs/provision_pe_xl_nodes.rb --noop --test
*** Running in no-op mode ***
*** Running in test mode ***

Would have provisioned pe_xl nodes with the following options:
  HA: false
  Output directory for Bolt inventory and parameter files: ./
  PE version: 2019.1.0
  AWS EC2 id tag: slv
  AWS EC2 instance type: c5.2xlarge
  AWS EC2 volume size: 80

Would have called:

  hosts = provision_hosts_for_roles(["master", "puppet_db", "compiler_a", "compiler_b"],
                                    slv,
                                    c5.2xlarge,
                                    80)

to provision the hosts, then:

  create_pe_xl_bolt_files(hosts, ./)

to create the Bolt inventory and parameter files.
```

But the output is not correct because in test mode it would not have called `provision_hosts_for_roles`.